impeller/aiks/aiks_unittests.cc

```diff
@@ -3362,5 +3362,31 @@ TEST_P(AiksTest, CaptureInactivatedByDefault) {
   ASSERT_FALSE(GetContext()->capture.IsActive());
 }
 
+// Regression test for https://github.com/flutter/flutter/issues/134678.
+TEST_P(AiksTest, ReleasesTextureOnTeardown) {
```

Comment:
Does this actually work on a "real" backend - does that matter? In theory if we've submitted a frame the fence waiter should keep the weak texture alive.

Reply:
Does (a) this test work on a real backend, (b) do we avoid the underlying issue on a real backend?

(a) No, at least I've not tried it. I just want to make sure we can enable validation layers for `aiks_unittests`.
     (https://github.com/flutter/flutter/issues/133506)

(b) This probably is, as you've mentioned, show a weakness in the fence waiter that needs to be fixed as well.

Reply:
LGTM, though we may need to adjust this further

Reply:
Oh, wait - your point is if the fence waiter was working properly we wouldn't need this :)?

Reply:
I think that if the fence waiter was working this may not be released by the end of the scope

Reply:
it would only be after Shutdown. So maybe manually call contextvk->Shutdown() and then check?

Reply:
Just to clarify, this test does fail (100% of the time) before our patch as-is.

Do you think I should add _another_ test or tweak this tests behavior?

Reply:
or, will become flaky after we fix fence waiter

Reply:
Gotcha, added a comment and ran 1000 times to ensure it's fine right now.